cmd/aws-vpc-cni-init/main.go

```diff
@@ -83,8 +79,11 @@ func configureSystemParams(procSys procsyswrapper.ProcSys, primaryIF string) err
 	// Note that older kernels may not support tcp_early_demux, so we must first check that it exists.
 	entry = "net/ipv4/tcp_early_demux"
 	if _, err := procSys.Get(entry); err == nil {
-		disableIPv4EarlyDemux := getEnv(envDisableIPv4TcpEarlyDemux, "false")
-		if disableIPv4EarlyDemux == "true" {
+		disableIPv4EarlyDemux, boolenv := utils.GetBoolAsStringEnvVar(envDisableIPv4TcpEarlyDemux, defaultDisableIPv4TcpEarlyDemux)
+		if boolenv != err {
+			log.WithError(boolenv).Errorf("Failed with incorrect environmental value")
+		}
```

Comment:
If the function returns an error, then the variable name should be `err`, not `boolenv`. The comparison at line 83 does not make any sense. Also, did you try passing the log to the function so that it does not have to return an error?

Whatever log you add should say something like: `"Failed to parse variable %s with value %s as boolean", envDisableIPv4TcpEarlyDemux, value`, which is why I think the logging makes more sense in the function

Reply:
Also, anywhere "environmental" is used needs to be changed to "environment"